File: assignments/a4/inklings/goink/internal/asciiart/ascii_art.go
package asciiart

import (
	"fmt"
	"time"

	"inklings/internal/sim"
)

// ANSI text colors (same codes as the original C++ TextColor enum)

type TextColor int

const (
	TextDefault TextColor = 39
	TextBlack   TextColor = 30
	TextRed     TextColor = 31
	TextGreen   TextColor = 32
	TextYellow  TextColor = 33
	TextBlue    TextColor = 34
	TextMagenta TextColor = 35
	TextCyan    TextColor = 36
	TextWhite   TextColor = 37
)

func setTextColor(c TextColor) {
	fmt.Printf("\033[%dm", int(c))
}

func resetTextColor() {
	setTextColor(TextDefault)
}

func clearTerminal() {
	fmt.Print("\033[H\033[J")
}

func DrawGridAndInklings(s *sim.State) {
	// TODO: draw ASCII grid and inklings based on s.Grid and s.Inklings.
	_ = s
}

func DrawState(s *sim.State) {
	// TODO: display ink levels, number of live goroutines, etc.
	_ = s
}

func EventLoop(s *sim.State) {
	// TODO: periodically redraw terminal and listen for input / commands.
	for {
		clearTerminal()
		DrawGridAndInklings(s)
		DrawState(s)
		time.Sleep(time.Second)
	}
}

File: assignments/a4/inklings/cink/src/ascii_art.c
#include "ascii_art.h"
#include <stdio.h>

void clear_terminal(void) {
    // Simple ANSI clear.
    printf("\033[H\033[J");
}

void draw_grid_and_inklings(const SimState *state) {
    // TODO: draw grid and inklings using ASCII.
    (void)state;
}

void draw_state(const SimState *state) {
    // TODO: display ink tank levels, number of live threads, etc.
    (void)state;
}

void event_loop(SimState *state) {
    // TODO: periodically redraw terminal, process keyboard input, etc.
    (void)state;
}


static void set_text_color(TextColor color) {
    printf("\033[%dm", (int)color);
}

static void reset_text_color(void) {
    set_text_color(TEXT_DEFAULT);
}